### PR TITLE
fix bugs in team reverse sigs and beef up testing

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: src/subkey.iced [hunk 1]
```diff
@@ -66,20 +65,28 @@ exports.SubkeyBase = class SubkeyBase extends Base
     await @reverse_sig_check { json, new_km: @get_new_km() }, esc defer()
     cb null
 
+  _get_reverse_sig : (json) ->
+    json?.body?[@get_field()]?.reverse_sig
+  _get_new_sibkid : (json) ->
+    json?.body?[@get_field()]?[@sibkid_slot()]
+  _clear_reverse_sig : (outer) ->
+    # body.sibkey.reverse_sig should be the only field different between the two
+    outer?.body?[@get_field()].reverse_sig = null
```
Comment (oconnor663) [inline src/subkey.iced]: Get rid of the `?`'s?

File: src/subkey.iced [hunk 2]
```diff
@@ -66,20 +65,28 @@ exports.SubkeyBase = class SubkeyBase extends Base
     await @reverse_sig_check { json, new_km: @get_new_km() }, esc defer()
     cb null
 
+  _get_reverse_sig : (json) ->
+    json?.body?[@get_field()]?.reverse_sig
```
Comment (oconnor663) [inline src/subkey.iced]: Since we're in here, should we call `get_field` something more descriptive?

File: src/team.iced [hunk 3]
```diff
@@ -28,15 +28,18 @@ class TeamBase extends SubkeyBase
...
   get_new_key_section : () -> @per_team_key
   set_new_key_section : (m) ->
     m.generation = @kms.generation
     m.encryption_kid = @kms.encryption.get_ekid().toString('hex')
+    m.signing_kid = @kms.signing.get_ekid().toString('hex')
     @per_team_key = m
   get_new_km : () -> @kms?.signing # use the signing KM
-  sibkid_slot : () -> "signing_kid"
-  need_reverse_sig : (json) -> json?.body?.per_team_key?
+  need_reverse_sig : (json) -> json?.body?.team?.per_team_key?
```
Comment (oconnor663) [inline src/team.iced]: It looks like we're not implementing `sibkid_slot` anymore. Is it ok that we inherit the parent implementation that returns `"kid"`? It looks like we're signing ourselves up to override all parent methods that call `sibkid_slot`, but that sounds like a tricky contract?

File: test/files/team.iced [hunk 4]
```diff
@@ -50,3 +52,42 @@ exports.test_bad_key_section = (T,cb) ->
...
+    cb null
+
+  await obj.generate_v2 esc defer out
+  typ = out.inner.obj.body.type
+  obj2 = alloc typ, arg
+  varg = { armored : out.armored, skip_ids : true, make_ids : true, inner : out.inner.str }
+  await obj2.verify_v2 varg, defer err
+  if corrupt
+    T.assert err?, "got an error back"
+    T.assert (err.message.indexOf('Reverse sig json mismatch') >= 0), "found right error message"
```
Comment (oconnor663) [inline test/files/team.iced]: Should we be asserting anything in the `!corrupt` case? It does look like we mean to test that case below.

**Comment by maxtaco** [comment]
Thanks for the review, all fixed!

**Comment by oconnor663** [comment]
LGTM thanks

**Pull request merged**